### ISSUE TITLE
Toolbar Search Remove My Android Back Press Handler 

### ISSUE THREAD

**Issue by adhicl** [xotahal/react-native-material-ui#203]
Hi, I have noticed a behavior problem with the toolbar that it will remove my own backhandler function.

So in my code I already add a BackHandler addEventListener to go back to previous screen.
When I add a searchable to the toolbar and click on it (the search bar become active) then I notice my previous BackHandler does not get called anymore.
If I don't click on the search bar, my BackHandler function can be called.
Can you help me with this?

PS: I read the code for toolbar.react.js and notice there is no removeEventListener for BackAndroid, but use remove() instead, maybe this is the cause?

**Comment by sunilkumarc** [comment]
Even I'm facing the same issue! When are you guys planning to fix this?

**Comment by aburai** [comment]
@adhicl the first backpress should end the search mode, the second backpress should trigger your own backhandler. This works for me.

The reason why the first backpress act like this is from the docs ->
// on android it's typical that back button closes search input on toolbar

**Comment by adhicl** [comment]
@aburai yes that is how I would want it. But in my case, it won't trigger my own backhandler as if it has been removed. So how I do this is by: add my own backhandler to backpress, set searchable, test run it, click on search, press back, search box closed, press back, my handler does not get called.
Can you provide me how you code this? Thanks before

**Comment by aburai** [comment]
@adhicl in my toolbar test component i do this

```javascript
  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.backPress);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.backPress);
  }

  backPress = () => {
    console.log('back pressed');
    return true;
  }
```

and then simply render a view with the searchable toolbar

```javascript
      <View style={{flex:1}}>
          <Toolbar
            leftElement="menu"
            centerElement="Searchable"
            searchable={{
              autoFocus: true,
              placeholder: 'Search'
            }}
          />
       </View>
```

this works for me.
